### PR TITLE
Sort icon

### PR DESCRIPTION
**Pull request by Taloth**
Sort icon now a bit more to the right and can overlap the header itself.
Also added tooltip support to the header.

Ow, and also fixed the sort direction cycle.

### CODE REVIEW

File: src/UI/Shared/Grid/HeaderCell.js [hunk 1]
```diff
@@ -17,13 +17,17 @@ module.exports = function() {
         render : function() {
             this.$el.empty();
             this.$el.append(this.column.get('label'));
+            if (this.column.get('tooltip')) {
+                this.$el.attr('title', this.column.get('tooltip'));
+                this.$el.attr('data-container', '.table');
```
Comment (kayone) [inline src/UI/Shared/Grid/HeaderCell.js]: use an object to set both at the same time, less touching the DOM = more faster.

**Comment by kayone** [comment]
can you post screenshots.

thanks.

File: src/UI/Content/theme.less [hunk 2]
```diff
@@ -292,3 +292,9 @@ dl.info {
     vertical-align : bottom;
   }
 }
+
+.sort-direction-icon {
+  .pull-right();
+  position: relative;
+  width: 0px;
```
Comment (markus101) [inline src/UI/Content/theme.less]: Are we aligning values in less files as well?

Comment (Taloth) [inline src/UI/Content/theme.less]: originally the pull-right was in the js file. I can move it back there if you want.

**Comment by Taloth** [comment]
Posted a couple of screenshots on hangout, so didn't think it was needed.
image

**Comment by markus101** [comment]
In before @kayone "CROPPED?!?!?!" :smile:

It looks good to me.

File: src/UI/Content/theme.less (hunk 2 above)
Comment (markus101) [inline src/UI/Content/theme.less]: Sorry, I meant the keys/values

```
position : relative;
width     : 0px;
```

File: src/UI/Shared/Grid/HeaderCell.js [hunk 3]
```diff
@@ -97,11 +101,16 @@ module.exports = function() {
             var column = this.column;
             var sortable = Backgrid.callByNeed(column.sortable(), column, collection);
             if (sortable) {
+                var isSorted = this.$el.children('.icon-sonarr-sort-asc,.icon-sonarr-sort-desc').length !== 0;
                 var direction = collection.state.order;
-                if (direction === 'ascending' || direction === -1) {
-                    direction = 'descending';
+                if (column.get('sortType') === 'fixed' || !isSorted) {
+                    direction = column.get('direction') || 'ascending';
                 } else {
```
Comment (markus101) [inline src/UI/Shared/Grid/HeaderCell.js]: New line the else (I'm royally confused with what our style should be at this point).

**Comment by kayone** [comment]
the red icon on the header could make people think there is something wrong.

**Comment by Taloth** [comment]
So you suggest making it black? add a icon to icons.less?

**Comment by kayone** [comment]
honestly, I would suggest to not have the icon, even in black (though it is better) it still suggests there is something wrong. any reason why not just have the blank column header with the sort indicator? if it's a tool tip we could just attach it to an invisible icon.

**Comment by Taloth** [comment]
We didn't have content in the header, I think we should. Having nothing in there caused the whole mess in the first place.

What about http://fortawesome.github.io/Font-Awesome/icon/exclamation/
Distinctly different than the http://fortawesome.github.io/Font-Awesome/icon/exclamation-circle/ we use in the rest of the application.

**Comment by kayone** [comment]
how about this, we add a green checkmark icon for the rows that don't have an error, and make the header http://fortawesome.github.io/Font-Awesome/icon/flag/

**Pull request merged**